client/src/components/relatedProducts/Carousel.jsx

```diff
@@ -9,31 +9,39 @@ import './Carousel.css'
 
 export default function Carousel(props) {
   const [relatedProducts, setRelatedProducts] = useState([])
-  const [productDetails, setProductDetails] = useState([])
+  const [currentProduct, setCurrentProduct] = useState([])
   const [currentIndex, setCurrentIndex] = useState(0)
 
 
-
   const currentRelated = relatedProducts.length
   const { productId } = useParams()
 
   useEffect(() => {
-    getRelatedProducts()
+    let mounted = true;
+    if (mounted) {
```

Comment:
A little confused what is going on here. This use effect only happens on mount but we are instantiating a variable 'mounted'. This variable never changes and is not effected by any changes below it. 

We then return a function that is also never used or called since it is inside a useEffect. I really feel like I am missing soemething here because I don't feel like you would have written these without a purpose.

Reply:
Oh yea, I removed that already but forgot to push it up.